### ISSUE TITLE
Support ListCollector

### ISSUE THREAD

**Issue by nojaf** [fable-compiler/Fable#3700]
### Description

It would be great if Fable could support [ListCollector<'T>](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-compilerservices-listcollector-1.html)

### Repro code

```fsharp
    let cutOffLast list =
        let mutable headList = ListCollector<'a>()

        let rec visit list =
            match list with
            | []
            | [ _ ] -> ()
            | head :: tail ->
                headList.Add(head)
                visit tail

        visit list
        headList.Close()
```

### Expected and actual results

Expected:

```js
// Something along the lines of
// let mutable headList = ListCollector<'a>()
const headList = [];

// headList.Add(head)
headList.push(head);

// headList.Close()
import { ofArray } from "fable-library/List.js";
ofArray(headList);
```

Actual: `error FABLE: Microsoft.FSharp.Core.CompilerServices.ListCollector`1.Close is not supported by Fable`.

### Related information

* Fable version: `dotnet fable --version`: 4.9.0
* Operating system: Windows

I might wanna dig into this if approved.

**Comment by MangelMaxime** [comment]
It seems to be a small enough type to write [Source](https://github.com/dotnet/fsharp/blob/712016d1754fbabc8c102777fb3a9e346351fc8f/src/FSharp.Core/seqcore.fs#L534-L593) so fo me it is ok to add it.

Because this is an F# type, what we do in general is we take ~heavy~ inspiration from the FSharp.Core implementation and include it inside of the fable-library folder. For example, this is what we did for [StringBuilder](https://github.com/fable-compiler/Fable/blob/1b40a3fd9509812582e1986cf82c54ecb49d1b26/src/fable-library/System.Text.fs).

The only things I have question about is that ListCollector is using `RuntimeHelpers.FreshConsNoTail` which is defined as:

```fs
let inline FreshConsNoTail head = head :: (# "ldnull" : 'T list #)
```

so it is emitting IL code and we will not be able to do that same goes for `SetFreshConsTail`. But I don't think this is blocking.

**Comment by ncave** [comment]
Closing as implemented by #3745, feel free to reopen if needed.
